src/content/reference/react/use.md

```diff
@@ -29,99 +29,120 @@ const value = use(resource);
-2. Next, <CodeStep step={2}>`storagePromise`</CodeStep> is passed to <CodeStep step={4}>`use`</CodeStep> to read the resolved value of the promise.
-3. Finally, the resolved value of the promise can now be used safely. This includes surfacing data read by <CodeStep step={4}>`use`</CodeStep> like <CodeStep step={5}>`storage.quota`</CodeStep> in the JSX returned by the component.
+1. First, the Promise returned by <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> is stored in the <CodeStep step={2}>`storagePromise`</CodeStep> state variable created by <CodeStep step={3}>`useState`</CodeStep>. This prevents the call to <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> and the Promise it returns from being recreated across renders.
+2. Next, <CodeStep step={2}>`storagePromise`</CodeStep> is passed to <CodeStep step={4}>`use`</CodeStep> to read the resolved value of the Promise.
+3. Finally, the resolved value of the Promise can now be used safely. This includes surfacing data read by <CodeStep step={4}>`use`</CodeStep> like <CodeStep step={5}>`storage.quota`</CodeStep> in the JSX returned by the component.
 
 <Pitfall>
 
-When passing promises to `use`, the promise must be cached with the `cache` function, a state variable, or context to prevent the promise from being recreated on every render.
+When passing Promises to `use`, the Promise must be cached with the `cache` function, a state variable, or context to prevent the Promise from being recreated on every render.
 
 ```js
 function Component() {
-  use(new Promise(...)) // ❌ promise will be recreated on every render
+  use(new Promise(...)); // ❌ Promise will be recreated on every render
   // ...
 ```
-instead, pass the promise to a state variable or context or wrap the function that returns the promise in `cache`:
+
+Instead, pass the Promise to a state variable or context or wrap the function that returns the Promise in `cache`:
 
 ```js
-import { use, useState, cache } from "react"
+import { use, useState, cache } from 'react';
 
 function Component() {
-  const [promise] = useState(() => new Promise(...)) // ✅ Promise persists across renders
-  const data = use(cache(fnThatReturnsAPromise())) // ✅ Promise persists across renders
-  //...
+  const [promise] = useState(() => new Promise(...)); // ✅ Promise persists across renders
+  const data = use(cache(fnThatReturnsAPromise())); // ✅ Promise persists across renders
+  // ...
+```
+
+Storing the results of a API call in a state variable, context, or cache means that value returned by `use` will not be updated if the underlying API returns a new value. If the underlying data changes you must call the API again and update the state variable or context to the new Promise returned by the API call:
+
+```js
+function StorageInfo() {
+  const [storagePromise, setStoragePromise] = useState(
```

Comment:
You changed the example above to use a callback here, does this need to be a callback too?

Reply:
yes! good catch, fixed.